packages/message-db/message-db/src/lib/Category.ts

```diff
@@ -199,7 +200,10 @@ export class MessageDbContext {
   }
 }
 
-type AccessStrategy<Event, State> =
+
+export type Project<State> = (conn: Client, streamId: Equinox.StreamId, state: State, version: bigint) => Promise<void>
```

Comment:
Project/projection is a very generic term
So is Index
But maybe it conveys "go off and make sure this is indexed" vs "maybe you want to project this somewhere somehow"
I guess I just really dislike Project
Maybe "ConsistentIndexing" or "InlineIndexing" ?

Reply:
"InlineProjection" works, as discussed on discord

Reply:
But then... That kind of implies a single projection and this thing is a giant-ass footgun.
Maybe `FootgunProjection` 😜

Reply:
CouplingHook?

Reply:
`UNSAFE_ProjectState<T>`

Reply:
`KimiProjection<T>`